src/Components/OATPropertyEditor/PropertyList.tsx

```diff
@@ -307,41 +320,43 @@ export const PropertyList = ({
                                         deleteItem={deleteItem}
                                         dispatch={dispatch}
                                         state={state}
-                                        setPropertyOnHover={setPropertyOnHover}
                                     />
                                 );
                             }
                         })}
                 </div>
-                <div
-                    className={
-                        propertyInspectorStyles.addPropertyBarPropertyListWrap
-                    }
-                    onMouseLeave={() => setPropertySelectorVisible(false)}
-                >
-                    {hover &&
-                        model &&
-                        model[propertiesKeyName].length > 0 &&
-                        !propertyOnHover && (
+                {propertyList && propertyList.length > 0 && (
+                    <div
+                        className={
+                            propertyInspectorStyles.addPropertyBarPropertyListWrap
+                        }
+                        onMouseLeave={(e) => {
+                            if (getIsMouseLeaveNotNorth(e)) {
+                                setPropertySelectorVisible(false);
+                            }
+                        }}
+                    >
+                        {true && (
                             <AddPropertyBar
-                                onMouseOver={() => {
+                                onMouseOver={(e) => {
                                     setPropertySelectorVisible(true);
                                     setLastPropertyFocused(null);
+                                    handleSelectorPosition(e);
```

Comment:
I'd prefer to extract this logic into a separate function rather than doing it inline.